modules/terraform-before-cluster.adoc

```diff
@@ -0,0 +1,133 @@
+// Module included in the following assemblies:
+//
+// * 
+
+:_content-type: PROCEDURE
+
+[id="terraform-before-cluster"]
+= Before creating a cluster
+The following use cases provide examples of actions taken before cluster creation.
+
+== Terraform prerequisites
+//modules/sd-terraform-prereqs.adoc
+//undo hiding the above include when Eric *finally* pushes 7235
+[id="terraform-iam-roles"]
+== Account-wide IAM roles
+Prior to creating a ROSA STS cluster, you must create the required account-wide roles and policies. See "Account-wide IAM role and policy reference" in the Additional resources for more information.
+
+.Prerequisites
+* You have an AWS account.
+* You have installed the latest version of the ROSA command-line (CLI) (`rosa`).
+* You have an offline OpenShift Cluster Manager token. This token can be generated in the Red Hat Hybrid Cloud Console link:https://console.redhat.com/[Red Hat Hybrid Cloud Console].
+* You have installed Terraform. See the link:https://developer.hashicorp.com/terraform/downloads[Terraform page]for the latest version.
+
+//.Open issue
+//This example also creates the policies for the Operator roles. When you update the version, it also updates the Operator role policies.
+
+.Procedure
+
+. To run `terraform apply` you need to set up some variables. This guide uses environmental variables. For more on Terraform variables, see link:https://developer.hashicorp.com/terraform/enterprise/workspaces/variables/managing-variables[Managing Variables] in the Terraform documentation.
++
+Run the following commands to export your variables. Provide your values in lieu of the brackets. Note that any values declared in the `variables.tf` file are the default values if you do not export a superseding value.
++
+.. This variable should be your full link:https://console.redhat.com/openshift/token[Openshift Cluster Manager offline token] that you generated in the prerequisites. 
++
+[source,terminal]
+----
+export TF_VAR_token=<ocm_offline_token>
+----
++
+.. This value should be the prefix for your Operator role.
++
+[source,terminal]
+----
+export TF_operator_role_prefix=<prefix_name>
+----
++
+.. This value should always point to `\https://api.openshift.com`.
++
+[source,terminal]
+----
+export TF_VAR_url=https://api.openshift.com
+----
++
+.. *Optional:* You can set the desired OpenShift version with this variable. The default is available from the ROSA CLI with `rosa list version |grep yes`. The OpenShift version should be in the form of x.y, for example 4.13.
++
+[source,terminal]
+----
+export TF_VAR_openshift_version=<choose_openshift_version>
+----
++
+.. *Optional:* If you want to set specific AWS tags for your account roles, you can use this variable to declare those tags.
++
+[source,terminal]
+----
+export TF_VAR_tags=<aws_resource_tags> (Optional)
+----
++
+
+. In your local copy of the `account-roles` folder, run the following command:
```

Comment:
We need to create some way to explain how to get to the Terraform modules. I think we need another section showing the `main.tf` file and talking about what we need to create X. In this case, we'll need a developer to explain what each line is doing, I think. We can talk to @nirarg for some ideas.